server/api/users.js

```diff
@@ -1,17 +1,80 @@
-const router = require('express').Router()
-const { models: { User }} = require('../db')
-module.exports = router
+const router = require('express').Router();
+const User = require('../db/models/User');
+const Order = require('../db/models/Order');
+
+const requireToken = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization;
+    const user = await User.findByToken(token);
+    req.user = user;
+    next();
+  }
+  catch (error) {
+    next(error);
+  }
+};
 
 router.get('/', async (req, res, next) => {
   try {
     const users = await User.findAll({
       // explicitly select only the id and username fields - even though
       // users' passwords are encrypted, it won't help if we just
       // send everything to anyone who asks!
-      attributes: ['id', 'username']
-    })
-    res.json(users)
-  } catch (err) {
-    next(err)
+      attributes: ['id', 'name', 'address', 'username'],
+    });
+    res.json(users);
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+router.post('/signup', async (req, res, next) => {
+  try {
+    const user = await User.authenticate(req.body);
+    const token = await user.generateToken();
+    res.json(token);
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+router.get('/:userId', requireToken, async (req, res, next) => {
+  try {
+    if (req.user.id === req.params.id) {
+      res.json(req.user);
+    }
+    else res.status(403).send();
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+router.get('/:userId/orders', requireToken, async (req, res, next) => {
```

Comment:
This route should actually confirm that the req.params.userId is the same as the req.user.id before making that database query